### PR TITLE
Introduce string split on editor

### PR DESCRIPTION
**Pull request by prabushi**
## Purpose
Fix https://github.com/ballerina-platform/plugin-vscode/issues/215

Related LS fix: https://github.com/ballerina-platform/ballerina-lang/pull/29808

### CODE REVIEW

**Comment by nadeeshaan** [comment]
Shall we tag the associated lang repo PR as well.

**Comment by prabushi** [comment]
> Shall we tag the associated lang repo PR as well.

Updated.

File: src/editor-support/split-provider.ts [hunk 1]
```diff
@@ -61,17 +61,19 @@ export class StringSplitter {
...
                 }
-                editor.edit(editBuilder => {
-                    const startPosition = new Position(range.start.line, range.start.character);
-                    const nextLinePosition = new Position(range.start.line + 1, documentChange!.text.length - 1);
-                    const endPosition = new Position(range.end.line + 1, documentChange!.text.indexOf('\n'));
-                    editBuilder.insert(startPosition, "\" +\n");
-                    editBuilder.insert(nextLinePosition, "\"");
-                    editBuilder.delete(new Range(startPosition, endPosition));
-                });
+                if (response.kind === 'STRING_LITERAL') {
```
Comment (nadeeshaan) [inline src/editor-support/split-provider.ts:67]: We can get the kind literals for constants

File: src/editor-support/split-provider.ts [hunk 2]
```diff
@@ -61,17 +61,19 @@ export class StringSplitter {
...
+                    editor.edit(editBuilder => {
+                        const startPosition = new Position(range.start.line, range.start.character);
+                        const nextLinePosition = new Position(range.start.line + 1, documentChange!.text.length - 1);
+                        const endPosition = new Position(range.end.line + 1, documentChange!.text.indexOf('\n'));
+                        editBuilder.insert(startPosition, "\" +\n");
```
Comment (nadeeshaan) [inline src/editor-support/split-provider.ts:72]: Shouldn't we make this platform specific line separator?

**Review by nadeeshaan — APPROVED** [review]

File: src/editor-support/split-provider.ts (hunk 1 above)
Comment (prabushi) [reply src/editor-support/split-provider.ts]: Updated.

File: src/editor-support/split-provider.ts (hunk 2 above)
Comment (prabushi) [reply src/editor-support/split-provider.ts]: Fixed & tested on Windows.

**Pull request merged**